### DISCUSSION TITLE
Need some help with the basics

### DISCUSSION THREAD

**Discussion by ATH4** [cc-tweaked/CC-Tweaked#1258] [Q&A] [answered]
so, my issue is that i cannot seem to figure out how to control a motor.

my motor is the electric motor from the mod "Create Crafts & additions".
this motor does support being controlled by a computercraft computer. (documentation here:  [](](https://github.com/mrh0/createaddition/blob/main/COMPUTERCRAFT.md))) )

my program is suppoed to check if a redstone wire connected to the computer's left side is turned on
if it is, then set the motor's(which is placed on top of the computer) speed to 256
and if its not on, set the speed to 0

according to my own googling, this should be fairly simple, but i could never get it to work, the usual errors about expecting end or = near something.

my code looks like this:
while true do
if redstone.getInput("Left") = 1 then
  motor = peripheral.wrap("left")
  motor.setSpeed(256)
else
  motor.stop
end

help would be really appriciated

**Comment by fatboychummy** [comment, marked as answer]
```lua
while true do
  if redstone.getInput("Left") = 1 then
    motor = peripheral.wrap("left")
    motor.setSpeed(256)
  else
    motor.stop
  end
```

`motor.stop` is missing (), so lua thinks you are trying to assign something to `motor.stop` but forgot to actually do `= something`

`motor.stop()` is what you want.

Also, you need one more `end`.

**Reply by fatboychummy** [reply]
Ah, you've also capitalized `Left`, which won't work either, and `motor` only needs to be wrapped once (pre-wrapped).

Final code should look like this:

```lua
local motor = peripheral.wrap("left")

while true do
  if redstone.getInput("left") = 1 then
    motor.setSpeed(256)
  else
    motor.stop()
  end
end
```

**Reply by ATH4** [reply]
Tried both applying the fixes myself and just pasting your finished code in.
both returned  " test.lua:4: 'then' expacted near '=' "

**Reply by fatboychummy** [reply]
Ah, missed another one. You need to use `==` rather than `=`. 

`=` sets values, `==` compares values.

You will probably want to do `== true` instead, or just remove the `==` part entirely, ie:

```lua
if redstone.getInput("left") then
```

Since `redstone.getInput` returns a boolean value, you don't need a comparison. `if x then` is pretty much equal to `if x == true then`

**Reply by ATH4** [reply]
appiled the fix, now it returns " test.lua:5: attempt to index local motor (a nil value) "

**Reply by fatboychummy** [reply]
Is the motor on the left side?

**Reply by ATH4** [reply]
IT WORKS! thank you so much.
after looking at the code i noticed it as looking  for a redstone signal from the same side the motor was on, changed that to right and bam, it worked as intended
